popup.js

```diff
@@ -49,26 +51,21 @@ document.addEventListener('DOMContentLoaded', function () {
 
 let activeButton = document.getElementById('active');
 
-activeButton.onclick = function(element) {
-  chrome.tabs.query({active: true, currentWindow: true}, function(tabs) {
-    let SPEED = document.getElementById('speed').value;
-    let DISBLED_USELESS_BUTTON = document.getElementById('disbled-btn').checked;
-    const code = `
-      document.getElementsByTagName('video')[0].playbackRate = ${SPEED};
+// Inject the script when start the tabs
+chrome.tabs.query({ active: true, currentWindow: true }, function(tabs) {
+  chrome.tabs.executeScript(tabs[0].id, { file: 'script.js' });
+});
 
-      if (${DISBLED_USELESS_BUTTON}) {
-        var list = document.querySelectorAll('vg-cuepoint.ng-scope');
-        
-        for (let item of list) {
-            item.style.display = 'none';
-        }
-      }
-      console.log('%c speed up ${SPEED * 100}% ! ', 'background: #222; color: #bada55');
-    `
+// Handle when user clicks the `active` button
+activeButton.onclick = function(element) {
+  chrome.tabs.query({ active: true, currentWindow: true }, function(tabs) {
+    let speed = document.getElementById('speed').value;
+    let isDisabledUselessButton = document.getElementById('disbled-btn')
+      .checked;
 
-    chrome.tabs.executeScript(
-        tabs[0].id,
-        {code});
+    chrome.tabs.sendMessage(tabs[0].id, {
```

Comment:
What's difference between `executeScript` and `sendMessage` ?

Reply:
executeScript just injected the script or code (that you wrote it before) to the `page` (page stands for the webpage that you open in the tab). 

`page` and `extension` use diferrent environment/scope. So that you can not pass the variable or anything directly. So that we use `sendMessage` as a message api to sent some obj to the tab or `page` as I said.

More example. It's look like `react-devtools`, `redux-devtools`. those libraries also use some how to inject their script to read the `page` and compute it and show it to use. That is a similar concept that we have.